server/src/main/java/org/apache/iotdb/db/metadata/idtable/AppendOnlyDiskSchemaManager.java

```diff
@@ -190,6 +198,65 @@ public Collection<DiskSchemaEntry> getAllSchemaEntry() throws IOException {
     return res;
   }
 
+  /**
+   * get DiskSchemaEntries from disk file
+   *
+   * @param offsets the offset of each record on the disk file
+   * @return DiskSchemaEntries
+   */
+  @Override
+  public List<DiskSchemaEntry> getDiskSchemaEntriesByOffset(List<Long> offsets) {
+    List<DiskSchemaEntry> diskSchemaEntries = new ArrayList<>(offsets.size());
+    Collections.sort(offsets);
+    try (RandomAccessFile randomAccessFile = new RandomAccessFile(dataFile, "r")) {
+      for (long offset : offsets) {
+        diskSchemaEntries.add(getDiskSchemaEntryByOffset(randomAccessFile, offset));
+      }
+    } catch (FileNotFoundException e) {
+      logger.info(e.getMessage());
+    } catch (IOException e) {
+      logger.error(e.getMessage());
+    }
+    return diskSchemaEntries;
+  }
+
+  /**
+   * delete DiskSchemaEntries on disk
+   *
+   * @param offset the offset of a record on the disk file
+   * @throws MetadataException
+   */
+  @Override
+  public void deleteDiskSchemaEntriesByOffset(long offset) throws MetadataException {
+    try (RandomAccessFile randomAccessFile = new RandomAccessFile(dataFile, "rw")) {
+      DiskSchemaEntry.writeTombstone(randomAccessFile, offset + FILE_VERSION.length() + 4);
+    } catch (IOException e) {
+      logger.error(e.getMessage());
+      throw new MetadataException(e.getMessage());
+    }
+  }
+
+  private DiskSchemaEntry getDiskSchemaEntryByOffset(RandomAccessFile randomAccessFile, long offset)
+      throws IOException {
+    randomAccessFile.seek(offset + FILE_VERSION.length() + 4);
```

Comment:
What's the meaning of this ```4```? Make this a global variable or a static variable.

Reply:
fixed